pkg/tiles/feature.go

```diff
@@ -0,0 +1,17 @@
+package tiles
+
+import connection "github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/connection"
+
+type FeatureType int64
+
+const (
+	NONE FeatureType = iota
+	ROADS
+	CITIES
+	FIELDS
+)
+
+type Feature struct {
+	FeatureType FeatureType
+	Connections []connection.Connection
```

Comment:
Hmm, it seems that we may not have the same idea about the design that we want to have - I was expecting `Feature` to be a specific *connection* (so we would get rid of `Connection` struct entirely) and have the following fields:
```
type Feature struct {
    Type FeatureType
    Sides []connection.Side
}
```
That would mean that the `Tile.Features` could contain multiple features of the same type and the functions such as `Cities()` would be generating a slice returning all features whose type matches the type that it's supposed to return.

@bstrzelecki did you want something like what I'm suggesting or does the current implementation match what you wanted already?

Reply:
But this way we wouldn't we able to differentiate connected and unconnected corners

Reply:
Why not?

![image](https://github.com/YetAnotherSpieskowcy/Carcassonne-Engine/assets/6032823/86c05671-f535-4312-abd5-25d5e0a86718)
```golang
Features: []Feature{
    {Type: CITY, Sides: []Side{connection.Top}},
    {Type: CITY, Sides: []Side{connection.Right}},
}
```

![image](https://github.com/YetAnotherSpieskowcy/Carcassonne-Engine/assets/6032823/ca61b01b-455e-4a9c-a7f5-2976fb31880a)
```golang
Features: []Feature{
    {Type: CITY, Sides: []Side{connection.Top, connection.Right}},
}
```

Reply:
:<

Reply:
changed